### PR TITLE
Button documentation

### PR DESCRIPTION
**Pull request by Charles-ErickBG**
### Proposed Changes

Added the documentation on the button component

### Potential Breaking Changes

None

### Acceptance Criteria

None

### CODE REVIEW

**Comment by Charles-ErickBG** [comment]
Wanted to add @jpmarceau (Doc team) as a reviewer but I'm not sure it worked. :/

**Comment by Charles-ErickBG** [comment]
Screen Shot 2019-09-18 at 2 47 34 PM
Screen Shot 2019-09-18 at 2 47 49 PM

**Comment by Charles-ErickBG** [comment]
I have integrated what we already had in the Styleguide as a Google Doc. The content will be improved over time. 

I also plan to fix the text styles as an extra PR in Vapor eventually.

File: packages/react-vapor/src/components/button/examples/ButtonExamples.tsx [hunk 1]
```diff
@@ -3,6 +3,7 @@ import {Svg} from '../../svg/Svg';
 import {Button} from '../Button';
 
 export class ButtonExamples extends React.Component<any, any> {
+    static description = 'Buttons are used to communicate and initialize actions when the user clicks them.';
```
Comment (jpmarceau) [inline packages/react-vapor/src/components/button/examples/ButtonExamples.tsx]: Suggestion:
Buttons communicate actions, and, when clicked, initialize those actions.

File: packages/react-vapor/src/components/button/examples/ButtonExamples.1.use.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
+## Guidelines
+
+### General
+
+-   **Use links**, instead, to **redirect users to another page** of the current website or app, or to another website.
```
Comment (jpmarceau) [inline packages/react-vapor/src/components/button/examples/ButtonExamples.1.use.md]: Suggestion:
To **redirect users to another page** of the current website or app, or to another website, **use links** rather than buttons.

File: packages/react-vapor/src/components/button/examples/ButtonExamples.1.use.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+-   Avoid using too many buttons on a page. If many buttons are to be added, **less popular** or **less important** actions may be **grouped in a “more” button** (dropdown list) or **visually styled as isolated actions or links**.
```
Comment (jpmarceau) [inline packages/react-vapor/src/components/button/examples/ButtonExamples.1.use.md]: Suggestion:
Avoid multiplying buttons within a given page. If you need several buttons, **less popular** or **less important** actions may be **grouped in a “more” button** (dropdown list) or **visually styled as isolated actions or links**.

I'm not sure what you mean by **visually styled as isolated actions or links** though? Maybe you could try to rephrase, and I can suggest something :)

File: packages/react-vapor/src/components/button/examples/ButtonExamples.1.use.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+### Labels
+
+-   The button **label** should allow to **foresee what will happen** when clicking it.
```
Comment (jpmarceau) [inline packages/react-vapor/src/components/button/examples/ButtonExamples.1.use.md]: allow to -> allow users to

File: packages/react-vapor/src/components/button/examples/ButtonExamples.1.use.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+-   Button **labels** should be **short** and use **“trigger words”**.
+-   As much as possible, the **first word of the label** should be an **action verb** (e.g., the label for a button that allows the user to add a filter could be “Add Filter").
+
+### Alterative Texts
+
+-   When a **button includes only an icon** (no label), the **label** should be added in the **alternative text element in the HTML** and appear in a **tooltip** when the cursor is above the button.
```
Comment (jpmarceau) [inline packages/react-vapor/src/components/button/examples/ButtonExamples.1.use.md]: (no label) -> (i.e., no label)
when the cursor is above the button -> when hovering the button

File: packages/react-vapor/src/components/button/examples/ButtonExamples.1.use.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+-   An **icon with a label does not need alternative text**. In this case, the icon is a visual help to identify the purpose faster and filling it will cause a screen reader to repeat the text twice.
```
Comment (jpmarceau) [inline packages/react-vapor/src/components/button/examples/ButtonExamples.1.use.md]: In this case, the icon is a visual help to identify the purpose faster and filling it will cause a screen reader to repeat the text twice.
->
In this case, the icon is a visual help to identify the purpose faster. Providing an alternative text would only repeat the information.

File: packages/react-vapor/src/components/button/examples/ButtonExamples.1.use.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+## Variations
+
+The Button Type choice is defined by the function of the button (e.g. primary or a secondary action), the number of actions available in the screen (e.g. only one, two or more actions) and its location (e.g. displayed in a header, action bar, page or footer).
```
Comment (jpmarceau) [inline packages/react-vapor/src/components/button/examples/ButtonExamples.1.use.md]: Suggestion:
The button type choice depends on the function of the button (e.g., primary action, secondary action), the number of actions available in the screen (e.g., one, two, several), and its location (e.g., header, action bar, page, footer).

File: packages/react-vapor/src/components/button/examples/ButtonExamples.1.use.md [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+| --------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| Primary   | Primary Buttons are used to enable the most important actions, that is those that lead to the completion of the main task (e.g., Save, Continue, Submit), or strong call to actions. Ideally, there should be only one Primary Button per screen to avoid dilluting the user's attention.                                                                                                                                                                                                  |
```
Comment (jpmarceau) [inline packages/react-vapor/src/components/button/examples/ButtonExamples.1.use.md]: Suggestion:
**Primary Buttons** enable the most important actions, i.e., those that lead to the completion of the main task (e.g., **Save**, **Continue**, **Submit**), or strong calls to action. Ideally, there should be only one **Primary Button** per screen to avoid diluting user attention.

File: packages/react-vapor/src/components/button/examples/ButtonExamples.1.use.md [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+| Secondary | Secondary Buttons are used for main actions that users might want to execute but that are not in direct relation to the main task that is to be done in the screen or interface.                                                                                                                                                                                                                                                                                                           |
```
Comment (jpmarceau) [inline packages/react-vapor/src/components/button/examples/ButtonExamples.1.use.md]: Suggestion:
Secondary Buttons are used for main actions -> **Secondary Buttons** enable main actions

File: packages/react-vapor/src/components/button/examples/ButtonExamples.1.use.md [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+| Combo     | The Combo Button got an attached drop menu because a choice of action has to be made (2 or more). They are usually used when one single action allows to create, delete or modify many different elements, which can be chosen in the dropdown list that appears on click. While using a combo button allows to avoid adding an extra step in the process, it can add friction to the user's decision making process if the options in the dropdown list are not intuitive enough to them. |
```
Comment (jpmarceau) [inline packages/react-vapor/src/components/button/examples/ButtonExamples.1.use.md]: Suggestion:
**Combo Buttons** have an attached drop menu where a choice between multiple actions has to be made (2 or more). Those buttons are useful when one single action allows users to create, delete or modify many different elements, which can be selected in the dropdown list that appears upon clicking. While using a **Combo Button** prevents adding an extra step in the process, it can add friction to the user decision making process when the options in the dropdown list are not intuitive enough.

File: packages/react-vapor/src/components/button/examples/ButtonExamples.1.use.md (hunk 2 above)
Comment (jpmarceau) [reply packages/react-vapor/src/components/button/examples/ButtonExamples.1.use.md]: There is a lot of bolding going on here, and I was wondering what is the idea behind it?
(I don't want to upset your practices, I'm mainly just curious because we're very parsimonious with bolding in the doc team generally :) )

File: packages/react-vapor/src/components/button/examples/ButtonExamples.1.use.md [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+| Isolated  | Isolated Action Buttons are used for extra actions located on their own within an interface. Button borders and background color are removed to reduce the noise generated by the presence of many consecutive borders and surfaces, or to make it more discrete in the screen. The icon is optional and aims to help users identify the action faster. However, keep in mind that too many icons in a screen can be as distracting and confusing as too less.                             |
```
Comment (jpmarceau) [inline packages/react-vapor/src/components/button/examples/ButtonExamples.1.use.md]: Suggestion:
**Isolated Action Buttons** enable extra actions which are, as the name suggests, isolated in an interface. Those button do not have borders and background colors, in order to reduce noise and to make them less conspicuous. Icons are optional and may help users identify actions faster. However, keep in mind that too many icons can be as distracting and confusing as too few.

File: packages/react-vapor/src/components/button/examples/ButtonExamples.1.use.md [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+| Icon Only | Icon Only Buttons are used when there is a lack of space and a high density of content. This button type is preferably used to represent labels that are easily represented by an icon, such as the word "More".                                                                                                                                                                                                                                                                           |
```
Comment (jpmarceau) [inline packages/react-vapor/src/components/button/examples/ButtonExamples.1.use.md]: Suggestion:
Icon Only Buttons are used when -> **Icon Only Buttons** are useful when

'easily represented by an icon, such as the word "More".' ? Do you mean that the word "More" is an icon? Or rather that it can easily be represented by an icon, such as "+" for instance? If I'm right, I would suggest:
that are easily represented by an icon, such as the word "More".   -> that are easily represented iconically, such as the icon "+" for "more items".

File: packages/react-vapor/src/components/button/examples/ButtonExamples.1.use.md (hunk 8 above)
Comment (otaillon) [inline packages/react-vapor/src/components/button/examples/ButtonExamples.1.use.md]:
```suggestion
| Primary   | Used to enable the most important actions,i.e., those that lead to the completion of the main task (e.g., Save, Continue, Submit), or strong call to actions. Ideally, there should be only one Primary Button per screen to avoid diluting user attention.                                                                                                                                                                                                  |
```

File: packages/react-vapor/src/components/button/examples/ButtonExamples.1.use.md (hunk 9 above)
Comment (otaillon) [inline packages/react-vapor/src/components/button/examples/ButtonExamples.1.use.md]:
```suggestion
| Secondary | Used for main actions that users might want to execute but that are not in direct relation to the main task that is to be done in the screen or interface.                                                                                                                                                                                                                                                                                                           |
```

File: packages/react-vapor/src/components/button/examples/ButtonExamples.1.use.md (hunk 11 above)
Comment (otaillon) [inline packages/react-vapor/src/components/button/examples/ButtonExamples.1.use.md]:
```suggestion
| Isolated  | Used for extra actions located on their own within an interface. Button borders and background color are removed to reduce the noise generated by the presence of many consecutive borders and surfaces, or to make it more discrete in the screen. The icon is optional and aims to help users identify the action faster. However, keep in mind that too many icons in a screen can be as distracting and confusing as too less.                             |
```

File: packages/react-vapor/src/components/button/examples/ButtonExamples.1.use.md (hunk 12 above)
Comment (otaillon) [inline packages/react-vapor/src/components/button/examples/ButtonExamples.1.use.md]:
```suggestion
| Icon Only | Used when there is a lack of space and a high density of content. This button type is preferably used to represent labels that are easily represented by an icon, such as the word "More".                                                                                                                                                                                                                                                                           |
```

File: packages/react-vapor/src/components/button/examples/ButtonExamples.tsx [hunk 13]
```diff
@@ -3,6 +3,7 @@ import {Svg} from '../../svg/Svg';
 import {Button} from '../Button';
 
 export class ButtonExamples extends React.Component<any, any> {
+    static description = 'Buttons are used to communicate and initialize actions when the user clicks them.';
     render() {
         return (
```
Comment (gdostie) [inline packages/react-vapor/src/components/button/examples/ButtonExamples.tsx]: While at it, it would be nice to add print delimiters to scope only the source code that is relevant to the example (i.e. what is inside the return parenthesis in the `render` method).

start delimiter: `// start-print`
stop delimiter: `// stop-print`

File: packages/react-vapor/src/components/button/examples/ButtonExamples.1.use.md [hunk 14]
```diff
@@ -0,0 +1,39 @@
...
+| Isolated  | Used for extra actions which are, as the name suggests, isolated in an interface. Those button do not have borders and background colors, in order to reduce noise and to make them less conspicuous. Icons are optional and may help users identify actions faster. However, keep in mind that too many icons can be as distracting and confusing as too few.                            |
+| Icon Only | Used when there is a lack of space and a high density of content. This button type is preferably used to represent labels that are easily represented by an icon, such as the word "More".                                                                                                                                                                                                                                                                           |
```
Comment (jpmarceau) [inline packages/react-vapor/src/components/button/examples/ButtonExamples.1.use.md]: ‘easily represented by an icon, such as the word "More".' ? Do you mean that the word "More" is an icon? Or rather that it can easily be represented by an icon, such as "+" for instance? If I'm right, I would suggest:
that are easily represented by an icon, such as the word "More". -> that are easily represented iconically, such as the icon "+" for "more items".

File: packages/react-vapor/src/components/button/examples/ButtonExamples.1.use.md [hunk 15]
```diff
@@ -0,0 +1,39 @@
+## Guidelines
+
+### General
+
+-   To redirect users to another page of the current website or app, or to another website, use links rather than buttons.
+-   Avoid multiplying buttons within a given page. If you need several buttons, less popular or less important actions may be grouped in a “more” button (dropdown list) or visually styled as isolated actions or links.
```
Comment (jpmarceau) [inline packages/react-vapor/src/components/button/examples/ButtonExamples.1.use.md]: I'm not sure what you mean by visually styled as isolated actions or links though? Maybe you could try to rephrase, and I can suggest something :)

File: packages/react-vapor/src/components/button/examples/ButtonExamples.1.use.md [hunk 16]
```diff
@@ -0,0 +1,39 @@
+## Guidelines
```
Comment (jpmarceau) [inline packages/react-vapor/src/components/button/examples/ButtonExamples.1.use.md]: So, as discussed, I've just tried to apply the doc review changes myself. I havent built the project to see how it renders though, and I've left two questions below. I also applied Olivier’s comments while I was at it. 

I’ve changed the bolding quite a lot, don’t hesitate if you dislike what I’ve done, I could change it back :)
I don't know what your bolding practices are in the UX team, and if you want to discuss I'm sure it would be beneficial to me.

File: packages/react-vapor/src/components/button/examples/ButtonExamples.1.use.md [hunk 17]
```diff
@@ -0,0 +1,39 @@
...
+
+---
+
+## Related Components
+
+[Action Bar](../#/components/ActionBar)
+
+["More" Menu Button](../#/components/Menu)
+
+[Tabs](../#/components/Tabs)
```
Comment (GermainBergeron) [inline packages/react-vapor/src/components/button/examples/ButtonExamples.1.use.md]:
```suggestion
[Tabs](#/components/Tabs)
```

File: packages/react-vapor/src/components/button/examples/ButtonExamples.1.use.md [hunk 18]
```diff
@@ -0,0 +1,39 @@
...
+| Isolated  | Used for extra actions which are, as the name suggests, isolated in an interface. Those button do not have borders and background colors, in order to reduce noise and to make them less conspicuous. Icons are optional and may help users identify actions faster. However, keep in mind that too many icons can be as distracting and confusing as too few.                            |
+| Icon Only | Used when there is a lack of space and a high density of content. This button type is preferably used to represent labels that are easily represented by an icon, such as the word "More".                                                                                                                                                                                                                                                                           |
+
+---
+
+## Related Components
+
+[Action Bar](../#/components/ActionBar)
+
+["More" Menu Button](../#/components/Menu)
```
Comment (GermainBergeron) [inline packages/react-vapor/src/components/button/examples/ButtonExamples.1.use.md]:
```suggestion
["More" Menu Button](#/components/Menu)
```

File: packages/react-vapor/src/components/button/examples/ButtonExamples.1.use.md [hunk 19]
```diff
@@ -0,0 +1,39 @@
...
+| Isolated  | Used for extra actions which are, as the name suggests, isolated in an interface. Those button do not have borders and background colors, in order to reduce noise and to make them less conspicuous. Icons are optional and may help users identify actions faster. However, keep in mind that too many icons can be as distracting and confusing as too few.                            |
+| Icon Only | Used when there is a lack of space and a high density of content. This button type is preferably used to represent labels that are easily represented by an icon, such as the word "More".                                                                                                                                                                                                                                                                           |
+
+---
+
+## Related Components
+
+[Action Bar](../#/components/ActionBar)
```
Comment (GermainBergeron) [inline packages/react-vapor/src/components/button/examples/ButtonExamples.1.use.md]: The link should always start with a # and be relative to the root
```suggestion
[Action Bar](#/components/ActionBar)
```

File: packages/react-vapor/src/components/button/examples/ButtonExamples.1.use.md (hunk 15 above)
Comment (GermainBergeron) [inline packages/react-vapor/src/components/button/examples/ButtonExamples.1.use.md]: `a ["more" button](../#/components/Menu) (dropdown list)`

File: packages/react-vapor/src/components/button/examples/ButtonExamples.1.use.md (hunk 2 above)
Comment (Charles-ErickBG) [reply packages/react-vapor/src/components/button/examples/ButtonExamples.1.use.md]: The principle that I like to use when using bold text is to help readers read faster by highlighting the important words. This way, readers can only read the bold text and understand the basic ideas behind the text.

TLDR style
